### ISSUE TITLE
Dockerfile:2 SC1072  : Windows hadolint installation

### ISSUE THREAD

**Issue by quixoticmonk** [hadolint/hadolint#436]
- [x] This is a bug report
- [ ] This is a feature request
- [ X] I searched existing issues before opening this one

### Expected behavior
Expected hadolint to review the dockerfile and respond with violations.

### Actual behavior
Error message : Dockerfile:2 SC1072  Fix any mentioned problems and try again. Checking the rules list, i couldn't find an SC1072
### Steps to reproduce the behavior

- On a windows 10 machine, install hadolint using scoop.
- Copy the dockerfile above.
- run the linter

**Output of `hadolint --version` or
  `docker run --rm hadolint/hadolint hadolint --version`:**
`Haskell Dockerfile Linter v1.17.7-0-gd807e87`
```bash
(paste your output here)
`Dockerfile:2 SC1072  Fix any mentioned problems and try again.``

**Dockerfile** (if relevant)

```Dockerfile
FROM openjdk:8-jdk-alpine
RUN addgroup -S spring && adduser -S spring -G spring
USER spring:spring
ARG JAR_FILE=target/*.jar
COPY ${JAR_FILE} app.jar
ENTRYPOINT ["java","-jar","/app.jar"]
```

**Additional environment details (OS, stack version, etc.)**
Windos 10

**Comment by zemanlx** [comment]
Thank you for reporting @quixoticmonk. Unfortunately, I am not able to replicate that behaviour. Here is how I have tried it in `bash`

```bash
$ docker run --rm -i hadolint/hadolint:v1.17.7 hadolint - << 'EOD'
FROM openjdk:8-jdk-alpine
RUN addgroup -S spring && adduser -S spring -G spring
USER spring:spring
ARG JAR_FILE=target/*.jar
COPY ${JAR_FILE} app.jar
ENTRYPOINT ["java","-jar","/app.jar"]
EOD
```

If you are able to replicate it on your computer can you please give us more information how you run it?

By the way, list of checks on README is not exhaustive and checks that start with SC are `shellcheck` checks that can be found on their wiki.

**Comment by quixoticmonk** [comment]
Thank you for the quick response @zemanlx.  
I have been running it on bash using 
![image](https://user-images.githubusercontent.com/11953035/83875209-b238dc00-a704-11ea-8afe-b9fb28bde418.png)

**Comment by zemanlx** [comment]
@quixoticmonk Interesting. Can you try to lint it using docker image? Can you share that exact file (I mean it can contain some characters that are not visible)?

**Comment by quixoticmonk** [comment]
Will try it on the CI and update here. Currently on a windows 10 machine which doesn't have docker properly setup.
[Dockerfile.txt](https://github.com/hadolint/hadolint/files/4736601/Dockerfile.txt)
Made it a .txt extension to upload here.

**Comment by zemanlx** [comment]
Mystery solved. It is good old CRLF vs LF, [the problem old as computers themself](https://en.wikipedia.org/wiki/Newline#History). @quixoticmonk If you format your file (its newlines) in *nix style it should all work.

@lorenzo this was introduced in `v1.17.7` with new parser. I have tested it with previous version a it works just fine. Can you please look into fixing this one too?

**Comment by phisco** [comment]
same error here on v1.18.0

**Comment by sharadpattanshetti** [comment]
I also receive same error as below linked to [:](https://github.com/hadolint/hadolint/issues/447#issue-653064968) 
PS C:\work\hadolint> wget  https://raw.githubusercontent.com/dockerfile/redis/master/Dockerfile |  docker run --rm -i hadolint/hadolint hadolint -
/dev/stdin:1:1 unexpected carriage return expecting '#', ADD, ARG, CMD, COPY, ENTRYPOINT, ENV, EXPOSE, FROM, HEALTHCHECK, LABEL, MAINTAINER, ONBUILD, RUN, SHELL, STOPSIGNAL, USER, VOLUME, WORKDIR, or end of input

**Comment by mloskot** [comment]
I'm experiencing the same issue using the hadolint-action v2.1.0 when linting `Dockerfile` based on Microsoft Windows ServerCore image and running some PowerShell comands:

![image](https://user-images.githubusercontent.com/80741/182463025-d0e6100d-0592-45f5-bfb9-dd907408d681.png)

The `Dockerfile:31` line is this:

```
RUN ["powershell", "-Command", "$ErrorActionPreference='Stop'; $ProgressPreference='SilentlyContinue'; Invoke-Expression ((New-Object System.Net.WebClient).DownloadString('https://chocolatey.org/install.ps1')); choco install -y --ignore-checksums vcredist-all; New-ItemProperty -Path Registry::HKEY_CLASSES_ROOT\\.css -Name 'Content Type' -PropertyType String -Value 'text/css'; New-ItemProperty -Path Registry::HKEY_CLASSES_ROOT\\.htm -Name 'Content Type' -PropertyType String -Value 'text/html'; New-ItemProperty -Path Registry::HKEY_CLASSES_ROOT\\.html -Name 'Content Type' -PropertyType String -Value 'text/html'; New-ItemProperty -Path Registry::HKEY_CLASSES_ROOT\\.js -Name 'Content Type' -PropertyType String -Value 'application/javascript';"]
```

**Comment by m-ildefons** [comment]
Current Hadolint (v2.14.0) works well with either CRLF (aka DOS) or LF (aka UNIX) style ASCII files.

<img width="880" height="183" alt="Image" src="https://github.com/user-attachments/assets/1fec6a65-f7e5-45b6-98f2-930d2d31333d" />

Therefore I'm going to close this issue.
